### ISSUE TITLE
Different probabilities for each realizations 

### ISSUE THREAD

**Issue by giulioturrisi** [do-mpc/do-mpc#54]
Hi,
is it possible to assign different probabilities for each scenario realizations? I cannot find any information in the doc!

Thank you

**Comment by 4flixt** [comment]
Hi @giulioturrisi,
this is currently not supported, all scenarios have the same probability. It is also no straight-forward how a scenario with a lower probability should be handled. Currently, all scenarios must satisfy all constraints. Would a scenario with lower probability be allowed to violate the constraints? It would be possible and will maybe be implemented in a future release to have "discount factors" for different scenarios. Their contribution to the cost function can then be reduced are increased. Is that what you had in mind in the first place?

**Comment by giulioturrisi** [comment]
Yes, I was thinking about the modification of the cost function to reduce the importance of a low probability scenario. I agree about the problem of the constraints satisfaction, maybe a "relaxation" of the constraints can be made available for an unlikely scenario.

**Comment by 4flixt** [comment]
This is a good point and will be taken into consideration! I'll rename this issue for a future implementation. Regarding the constraint relaxation: We already have this implemented through the set_nl_cons method. This can configure to be a "soft constraint". But this also works on all scenarios.
